diff --git a/src/libtools/signals.c b/src/libtools/signals.c
index ababbda..ceeb3b1 100644
--- a/src/libtools/signals.c
+++ b/src/libtools/signals.c
@@ -1776,9 +1776,9 @@ EXPORT void my_makecontext(x64emu_t* emu, void* ucp, void* fnc, int32_t argc, in
     int regs_abi[] = {_DI, _SI, _DX, _CX, _R8, _R9};
     for (int i=0; i<argc; ++i) {
         // get value first
-        uint32_t v;
+        uint64_t v;
         if(n<6)
-            v = emu->regs[regs_abi[n++]].dword[0];
+            v = emu->regs[regs_abi[n++]].q[0];
         else
             v = argv[j++];
         // push value
